src/ObjectPromiseProxy.js

```diff
@@ -56,44 +35,82 @@ function ObjectPromiseProxy (promise, target) {
           // 500 doesn't return a parsable response
         }
         // TODO: add all errors from the API response to the target
-        target.errors = {
-          ...target.errors,
+        // also TODO: split server errors by target once the info is available from the API
+        // maybe handle not just the first target lol
+        targets[0].errors = {
+          ...targets[0].errors,
           status: status,
           base: [{ message: message }],
           server: json.errors
         }
 
-        const errorString = JSON.stringify(target.errors)
+        const errorString = JSON.stringify(targets[0].errors)
         return Promise.reject(new Error(errorString))
       }
     },
     function (error) {
-      // TODO: Handle error states correctly
-      target.isInFlight = false
+      // TODO: Handle error states correctly, including handling errors for multiple targets
+      targets.forEach(t => { t.isInFlight = false })
       target.errors = error
       throw error
       // return target
     }
   )
   // Define proxied attributes
-  const attributeNames = Object.keys(target.attributeNames)
+  const attributeNames = Object.keys(targets[0].attributeNames)
   const tempProperties = attributeNames.reduce((attrs, key) => {
     attrs[key] = {
-      value: target[key],
+      value: targets[0][key],
       writable: false
     }
     return attrs
   }, {})
 
+  // TODO: how is isInFlight used? how should it be set for multiple targets?
   Object.defineProperties(result, {
-    isInFlight: { value: target.isInFlight },
+    isInFlight: { value: targets[0].isInFlight },
     ...tempProperties
   })
```

Comment:
these lines can be removed completely

Reply:
Because `isInFlight` isn't useful? Even if we aren't using it in portal, it does seem ostensibly useful (and this package is ostensibly open source). It also doesn't seem too tricky to answer your `TODO` question above. It should be `true` for each target in a bulk request while the request is in flight, and false when we receive a response. They are all sent/received at the same time, so I think it's pretty straightforward, but maybe I am misunderstanding the question in the comment?

Reply:
the original `TODO` comment was about `isInFlight`, but we would also need to cover the `...tempProperties`. For each individual target this is straightforward but what about the properties that we are adding to the promise as a whole? What consumes these values and how then should we deal with a single target vs many? Should it always be an array even if the length is 1? And when (if ever) should `isInFlight` be removed or updated from the result?

You raise a good point that this is ostensibly open source so perhaps we should support it, but right now I don't think it's documented and @jtorreggiani's original suggestion was to remove it for the time being (hence my comment to just get rid of this section).